### ISSUE TITLE
Lighthouse is sending `exited` validators to MEV-Boost but should only send `active` or `pending` ones

### ISSUE THREAD

**Issue by jclapis** [sigp/lighthouse#3465]
## Description

I recently set up MEV-Boost for Goerli/Prater. I have several exited validators on Prater (such as https://prater.beaconcha.in/validator/0x8aff2971a436a9a8783fb5574f923bb70b06f74e155961f1b448be0a733230ac5b600e87be1db201ba733cfce57b4660 which was used for slash testing) and during Lighthouse's validator registration process, it sends this validator over to MEV-Boost which causes it to fail.

According to [the official spec](https://ethereum.github.io/builder-specs/#/Builder/registerValidator), only `active` or `pending` validators should be sent to the builder during validator registration.

Here are MEV-Boost's logs:
```
mev-boost_1      | time="2022-08-12T17:56:48Z" level=info msg="http: POST /eth/v1/builder/validators 200" duration=0.182599993 method=POST module=service path=/eth/v1/builder/validators status=200
mev-boost_1      | time="2022-08-12T17:56:51Z" level=warning msg="error calling registerValidator on relay" error="HTTP error response: 400 / {\"code\":400,\"message\":\"not a known validator: 0x8aff2971a436a9a8783fb5574f923bb70b06f74e155961f1b448be0a733230ac5b600e87be1db201ba733cfce57b4660\"}\n" method=registerValidator module=service numRegistrations=36 ua= url="https://builder-relay-goerli.flashbots.net/eth/v1/builder/validators?id=rocketpool"
mev-boost_1      | time="2022-08-12T17:56:51Z" level=info msg="http: POST /eth/v1/builder/validators 502" duration=0.101773263 method=POST module=service path=/eth/v1/builder/validators status=502
```

Here are Lighthouse's BN logs:
```
eth2_1           | Aug 12 17:56:51.456 INFO Forwarding register validator request to connected builder, count: 36
eth2_1           | Aug 12 17:56:51.560 ERRO Error from connected relay              error: ServerMessage(ErrorMessage { code: 502, message: "no successful relay response", stacktraces: [] })
eth2_1           | Aug 12 17:56:51.560 WARN Error processing HTTP API request       method: POST, path: /eth/v1/validator/register_validator, status: 500 Internal Server Error, elapsed: 108.337824ms
```

I [reported this to Flashbots](https://github.com/flashbots/mev-boost/issues/245), who pointed me to the builder API spec above.

This issue is also present in [Nimbus](https://github.com/status-im/nimbus-eth2/issues/3961) and [Teku](https://github.com/ConsenSys/teku/issues/6070).

## Version
v2.5.1

## Present Behaviour
Lighthouse appears to be sending all of the connected validators to the builder 

## Expected Behaviour
Lighthouse should only register validators that are `active` or `pending`

## Steps to resolve
Filter the validators sent to the builder registration route so only `active` or `pending` ones are sent

**Comment by xrchz** [comment]
This is closed by #3473 right?

**Comment by michaelsproul** [comment]
Yeah good catch, ty!
